canonical/maas-ui
# feat: Configurable session timeout MAASENG-1121

## Done

- Added page to Security settings to configure session length
- Added selector to fetch `session_length` from backend
- Added tests for new component and selector

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Navigate to `/settings/security/session-timeout`
- Change the value of the session timeout (e.g. 1 week 5 days) and submit
- Reload the page
- **Ensure the value updates**
- Repeat this with different values, ensuring the above
- Try to break the form and ensure bad values cannot be submitted (e.g. 20 millenia and 12 picoseconds)

## Fixes

Fixes https://warthogs.atlassian.net/browse/MAASENG-1121

## Screenshots

![image](https://user-images.githubusercontent.com/35104482/211595579-84b3e266-d78e-45f3-851a-689b9f075b58.png)
![image](https://user-images.githubusercontent.com/35104482/211595649-b8da3d6f-3956-482f-bb97-6fab3764fab4.png)
